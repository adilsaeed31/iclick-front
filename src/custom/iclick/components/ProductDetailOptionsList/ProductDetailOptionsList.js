import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";
import { observer } from "mobx-react";
import BadgeOverlay from "@reactioncommerce/components/BadgeOverlay/v1";
import { BADGE_LABELS, BADGE_TYPES, badgeStatus } from "@reactioncommerce/components/BadgeOverlay/v1/utils";
import ProductDetailOption from "custom/iclick/components/ProductDetailOption";

@observer
export default class OptionsList extends Component {
  static propTypes = {
    onSelectOption: PropTypes.func,
    options: PropTypes.arrayOf(PropTypes.object),
    productSlug: PropTypes.string,
    selectedOptionId: PropTypes.string,
    theme: PropTypes.object
  }

  renderBadges(option) {
    const status = badgeStatus(option, BADGE_LABELS);

    if (!status) return null;

    return (
      <div>
        <BadgeOverlay
          product={option}
          filterOnly={BADGE_TYPES.SALE}
          shouldShowPrimaryOnly={true}
          label={status.label}
        />
      </div>
    );
  }

  render() {
    const { onSelectOption, options, selectedOptionId, theme } = this.props;

    if (!Array.isArray(options)) return null;

    return (
      <div className="d-flex flex-wrap">
        {options.map((option) => (
          <Fragment>
            <ProductDetailOption isActive={selectedOptionId === option._id} onClick={onSelectOption} option={option} />
            {this.renderBadges(option)}
          </Fragment>
        ))}
      </div>
    );
  }
}
