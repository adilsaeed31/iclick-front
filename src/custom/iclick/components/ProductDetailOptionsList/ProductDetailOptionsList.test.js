import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import { Provider } from "mobx-react";
import theme from "custom/reactionTheme";
import options from "./__mocks__/options.mock";
import ProductDetailOptionsList from "./ProductDetailOptionsList";

const uiStore = {
  pdpSelectedOptionId: "cmVhY3Rpb24vY2F0YWxvZ1Byb2R1Y3RWYXJpYW50OkpobjJ0YzVaRkFzYVdiaXZR"
};

test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider uiStore={uiStore}>
        <ProductDetailOptionsList
          options={options}
        />
      </Provider>
    </MuiThemeProvider>));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
