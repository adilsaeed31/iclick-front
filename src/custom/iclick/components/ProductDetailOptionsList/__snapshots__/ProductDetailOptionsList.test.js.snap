// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiGrid-container-4 MuiGrid-spacing-xs-8-26 SkOptionsList-root-1"
>
  <div
    className="MuiGrid-item-5"
  >
    <button
      className="MuiButtonBase-root-105 SkProductDetailOption-optionButton-101"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiTypography-root-108 MuiTypography-body1-117 SkProductDetailOption-optionText-102"
      >
        Blue
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-item-5"
  >
    <button
      className="MuiButtonBase-root-105 SkProductDetailOption-optionButton-101"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiTypography-root-108 MuiTypography-body1-117 SkProductDetailOption-optionText-102"
      >
        Red
      </span>
    </button>
  </div>
</div>
`;
